src/app/containers/NewSignupContainer/useSignup.js

```diff
@@ -0,0 +1,157 @@
+        planName: initialPlan,
+        cycle: CYCLE.YEARLY,
+        email: '',
+        username: '',
+        password: '',
+        currency: defaultCurrency
+    });
+
+    /**
+     * Verifies if payment was done and saves payment details for signup
+     * @param {*=} paymentParameters payment parameters from usePayment
+     * @returns {Promise<{ VerifyCode, paymentParameters }>} - paymentDetails
+     */
+    const checkPayment = async (model, paymentParameters) => {
+        const selectedPlan = getPlan(model.planName, model.cycle, plans);
+        const amount = selectedPlan.price.total;
+
+        if (amount > 0) {
+            const { VerifyCode } = await api(
+                verifyPayment({
+                    Amount: amount,
+                    Currency: model.currency,
+                    ...paymentParameters
+                })
+            );
+
+            return { VerifyCode, paymentParameters };
+        }
+
+        return null;
+    };
+
+    const getToken = ({ appliedCoupon, invite, verificationToken, paymentDetails }) => {
+        if (invite) {
+            return { Token: `${invite.selector}:${invite.token}`, TokenType: 'invite' };
+        } else if (appliedCoupon) {
+            return { Token: appliedCoupon.Coupon.Code, TokenType: 'coupon' };
+        } else if (paymentDetails) {
+            return { Token: paymentDetails.VerifyCode, TokenType: 'payment' };
+        }
+        return verificationToken;
+    };
+
+    // TODO: appliedCoupon (and gift code?)
+    const signup = async (model, signupToken) => {
+        const { Token, TokenType } = getToken(signupToken);
+        const { planName, password, email, username, currency, cycle } = model;
+        const selectedPlan = getPlan(planName, cycle, plans);
+
+        await srpVerify({
+            api,
+            credentials: { password },
+            config: queryCreateUser({
+                Token,
+                TokenType,
+                Type: CLIENT_TYPE,
+                Email: email,
+                Username: username
+            })
+        });
+
+        const { UID, EventID, AccessToken, RefreshToken } = await srpAuth({
+            api,
+            credentials: { username, password },
+            config: auth({ Username: username })
+        });
+
+        // Add subscription
+        // Amount = 0 means - paid before subscription
+        if (planName !== PLAN.FREE) {
+            const subscription = {
+                PlanIDs: {
+                    [selectedPlan.id]: 1
+                },
+                Amount: 0,
+                Currency: currency,
+                Cycle: cycle
+                // CouponCode: appliedCoupon ? appliedCoupon.Coupon.Code : undefined
+            };
+            await api(withAuthHeaders(UID, AccessToken, subscribe(subscription)));
+        }
+
+        // Add payment method
+        if (signupToken.paymentDetails) {
+            const { Payment } = await handle3DS(
```

Comment:
I recently changed the logic in [this PR](https://github.com/ProtonMail/react-components/pull/242). So you need to do something like this:

``` jsx
const { Payment } = await handle3DS({
    params: {
        Amount: selectedPlan.price.total,
        Currency: currency,
        ...signupToken.paymentDetails.paymentParameters
    },
    api,
    createModal
});
```